src/util/isPromise.ts

```diff
@@ -1,3 +1,3 @@
-export function isPromise(value: any): boolean {
-  return value && typeof value.subscribe !== 'function' && typeof value.then === 'function';
-}
\ No newline at end of file
+export function isPromise<T>(value: any | Promise<T>): value is Promise<T> {
```

Comment:
The return type still behaves as a boolean, but it also acts as a type guard as well.

Given the following:

    var value: Observable<T> | Promise<T>
    if (isPromise(value)) {
        // to the type system value is now just Promise<T>
    } else {
        // to the type system value is now just Observable<T>
    }